src/v/pandaproxy/schema_registry/json.cc

```diff
@@ -1111,35 +1111,6 @@ bool is_object_properties_superset(
     return true;
 }
 
-bool is_object_pattern_properties_superset(
```

Comment:
that's unfortunate. I find myself wondering whether it would make sense to implement some idea of "post-migration" compat features, the idea being that once an end user has cleared the bar of importing their old schemas to redpanda SR, we can switch on checks like this.

no clue whether it's worth the effort from a product perspective, but it feels icky to be stuck omitting things like this in perpetuity.

Reply:
yeah i think so too. but i realize users of schema registry use client serde libraries to produce, so really we are trying to match the legacy software and the client libraries that are generating these schemas

Reply:
sure, but is the client really opinionated about whether two particular subject versions are compatible? seems like as long as the wire format is implemented correctly we (redpanda SR) could fix these omissions on the backend. anyway, food for thought maybe. nothing to do here.